src/stories/components/custom-popover/custom-popover.tsx

```diff
@@ -30,20 +33,32 @@ export const CustomPopover = ({
     horizontal: 'center',
   }, ...props
 }: CustomPopoverProps) => {
+  const dispatch = useAppDispatch();
+  const state = store.getState();
+  const idPopoverOpenState = getPopoverOpen(state);
   const isLight = useThemeContext().themeMode === 'light';
   const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
   const [leaveTimeout, setLeaveTimeout] = React.useState<NodeJS.Timeout>();
+  const [openIdPopover, setOpenIdPopover] = useState('');
 
-  const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+  const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>, id: string) => {
+    setOpenIdPopover(id);
```

Comment:
Please, note that you are already storing the popup id in redux store, it is needed to have a different state here instead of just select that id from the redux store?